Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/CameraReelGalleryController.cs

```diff
@@ -0,0 +1,467 @@
+                    (cameraReelResponse, sprite) => reelThumbnailCache.Add(cameraReelResponse, sprite),
+                    cameraReelResponse => ThumbnailClicked?.Invoke(cameraReelResponse));
+
+                for (int i = 0; i < thumbnailViews.Count; i++)
+                    thumbnailImages[currentSize + i] = thumbnailViews[i];
+
+                currentSize += thumbnailViews.Count;
+            }
+            endVisible = currentSize - 1;
+
+            //Wait for layout to update after the addition of new elements
+            await UniTask.WaitWhile(() => Mathf.Approximately(view.verticalScrollbar.handleRect.rect.height, handleHeight) && currentSize > MINIMUM_ELEMENTS_FOR_WAITING_LAYOUT, cancellationToken: ct);
+
+            HandleElementsVisibility(ScrollDirection.UP);
+
+            previousY = view.scrollRect.verticalNormalizedPosition;
+            isLoading = false;
+        }
+
+        private void ScrollBeginDrag() => isDragging = true;
+
+        private void ScrollEndDrag()
+        {
+            isDragging = false;
+            CheckNeedsToLoadMore();
+        }
+
+        private void OnScrollRectValueChanged(Vector2 value)
+        {
+            HandleElementsVisibility(value.y > previousY ? ScrollDirection.UP : ScrollDirection.DOWN);
+            CheckNeedsToLoadMore();
+
+            previousY = value.y;
+
+            if (endVisible < beginVisible)
+                ResetThumbnailsVisibility();
+        }
+
+        private void ResetThumbnailsVisibility()
+        {
+            beginVisible = -1;
+            for (int i = 0; i < currentSize; i++)
+                if (ViewIntersectsImage(thumbnailImages[i].view.thumbnailImage))
+                {
+                    if (beginVisible < 0)
+                        beginVisible = i;
+
+                    EnableThumbnailImage(thumbnailImages[i]);
+                    endVisible = i;
+                } else
+                    DisableThumbnailImage(thumbnailImages[i]);
+        }
+
+        private void CheckNeedsToLoadMore()
+        {
+            if (currentSize - endVisible < view.loadMoreCounterThreshold && !pagedCameraReelManager.AllImagesLoaded && !isLoading && !isDragging)
+                LoadMorePageAsync(loadNextPageCts.Token).Forget();
+        }
+
+        private void DisableThumbnailImage(ReelThumbnailController thumbnailController)
+        {
+            thumbnailController.view.thumbnailImage.sprite = null;
+            thumbnailController.view.thumbnailImage.enabled = false;
+        }
+
+        private void EnableThumbnailImage(ReelThumbnailController thumbnailController)
+        {
+            if (reelThumbnailCache.TryGetValue(thumbnailController.CameraReelResponse, out Sprite sprite))
+                thumbnailController.view.thumbnailImage.sprite = sprite;
+            thumbnailController.view.thumbnailImage.enabled = true;
+        }
+
+        private void HandleElementsVisibility(ScrollDirection scrollDirection)
```

Comment:
I tried initially with that but the problem is that the LoopListView pools the objects messing up the layout since every item has different height and there is a vertical layout in the `Content` parent.